src/core/shape/2d_primitives.js

```diff
@@ -375,7 +375,18 @@ p5.prototype.quad = function() {
   p5._validateParameters('quad', arguments);
 
   if (this._renderer._doStroke || this._renderer._doFill) {
-    this._renderer.quad.apply(this._renderer, arguments);
+    if (this._renderer.isP3D && arguments.length !== 12) {
+      // if 3D and we weren't passed 12 args, assume Z is 0
+      // prettier-ignore
+      this._renderer.quad.call(
+        this._renderer,
+        arguments[0], arguments[1], 0,
+        arguments[2], arguments[3], 0,
+        arguments[4], arguments[5], 0,
+        arguments[6], arguments[7], 0);
+    } else {
+      this._renderer.quad.apply(this._renderer, arguments);
```

Comment:
Perhaps also put something in the description above. Something to the effect of "z-arguments only work when quad() is used in WEBGL mode"

Reply:
ok hope I put it in the right place.